drag_and_drop_v2/public/js/drag_and_drop.js

```diff
@@ -1,69 +1,114 @@
-function DragAndDropBlock(runtime, element) {
-
+function DragAndDropBlock(runtime, element, configuration) {
+    "use strict";
+    // Ensure "undefined" has not been redefined (though this unlikely and often impossible).
+    // Now we can check for 'undefined' using just '=== undefined' throughout this file.
+    if (undefined !== void 0) { console.log("WARNING: 'undefined' redefined"); var undefined = void 0; }
     // Set up gettext in case it isn't available in the client runtime:
-    if (typeof gettext == "undefined") {
+    if (gettext === undefined) {
         window.gettext = function gettext_stub(string) { return string; };
     }
 
-    var root = $(element).find('.xblock--drag-and-drop')[0];
+    var $element = $(element);
+    // root: root node managed by the virtual DOM
+    var root = $element.find('.xblock--drag-and-drop')[0];
+    var $root = $(root);
```

Comment:
@bradenmacdonald Nit: Unless there's a specific reason to initialize the variables in this order, the following would be slightly more intuitive I think:

```js
    var $root = $element.find('.xblock--drag-and-drop');
    var root = root[0];
```